File: 2656-maximum-sum-with-exactly-k-elements/2656-maximum-sum-with-exactly-k-elements.cpp
class Solution {
public:
  int maximizeSum(vector<int> &nums, int k) {

    int ans = 0;
    int mx = 0;
    for (auto &it : nums)
      mx = max(mx, it);

    int n = mx + k - 1;
    int tot = (n * (n + 1)) / 2;
    int rn = mx - 1;
    int rtot = (rn * (rn + 1)) / 2;

    return tot - rtot;
  }
};

File: 2656-maximum-sum-with-exactly-k-elements/2656-maximum-sum-with-exactly-k-elements.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var maximizeSum = function (nums, k) {

    var mx = Number(0), ans = Number(0);

    for (let val of nums)
        mx = Math.max(mx, val);

    var n = Number(mx + k - 1);
    var tot = Number((n * (n + 1)) / 2);
    var rn = mx - 1;
    var rtot = Number((rn * (rn + 1)) / 2);

    return tot - rtot;
};